### DISCUSSION TITLE
how to access int references of ssl_session_st struct in open SSL 3.1.3  version

### DISCUSSION THREAD

**Discussion by nagu1137** [openssl/openssl#22748] [Q&A] [answered]
we are upgrading from 1.0.2Q  to 3.1.3.

can you please help us with fllowing query.

how to access int references of ssl_session_st struct in open SSL 3.1.3  version

**Comment by t8m** [comment, marked as answer]
Unfortunately you cannot directly access them. The members of the opaque structs in openssl are internal and not accessible by applications since 1.1.0 version.

For the members which are generally useful to applications there were accessor functions added.
See the SSL_SESSION_get* and SSL_SESSION_set* functions in include/openssl/ssl.h and in the manual pages.

**Reply by nagu1137** [reply]
thanks for the response .. i dont find any SSL_SESSION_set function for references variable. any clue for that/?

**Reply by t8m** [reply]
For refcounting you need to use 
```
int SSL_SESSION_up_ref(SSL_SESSION *ses);
void SSL_SESSION_free(SSL_SESSION *ses);
```

**Comment by nagu1137** [comment]
I need to change the reference count .how can I do that?

Nagaraju

**Reply by t8m** [reply]
You can only increase it by one by using up_ref() or decreasing by one by free().
